discojs/discojs-core/src/dataset/data/tabular_data.ts

```diff
@@ -21,7 +21,8 @@ export class TabularData extends Data {
     try {
       await dataset.iterator()
     } catch (e) {
-      throw new Error('Data input format is not compatible with the chosen task')
+      console.error('Data input format is not compatible with the chosen task.')
+      throw (e)
```

Comment:
good idea to actuall keep the error we got; we can now even [wrap the received error in the thrown `Error` via cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause). now it can serve as context around the received error (as in most modern languages). also, avoiding console.* let users of the library choose how/when to print errors (such as a line in a webapp instead of the console). 

```suggestion
      throw new Error('Data input format is not compatible with the chosen task.', { cause: e })
```

Reply:
Ah great I googled a bunch for that and didn't find this syntax thanks

Reply:
Okay so lib ES2022 is needed to compile an error with cause and I couldn't do it without raising new issues. I'll let you change it if you want as I'm not sure of what upgrading to ES2022 entails.

Reply:
ho, I didn't check which ECMA version it was definied in, too bad; yeah, I'll try to update it later then, thanks for point this out.